File: JavaScript_Courses/JavaScript_Basics/Preparation/05_While_Loop_Exercise/04_Walking.js
function walking(input) {
    let targetSteps = 10000;
    let goingHomeCommand = "Going home";

    let sumSteps = 0;
    let i = 0;

    let currentAction = input[i];
    while (currentAction != goingHomeCommand) {
        let currentSteps = Number(currentAction);
        sumSteps += currentSteps;

        if (sumSteps >= targetSteps) {
            console.log("Goal reached! Good job!");
            console.log(`${sumSteps - targetSteps} steps over the goal!`);
        }

        i++;
        if (i >= input.length) {
            break;
        }
        currentAction = input[i];
    }

    if (currentAction == goingHomeCommand) {
        sumSteps += Number(input[i + 1]);
        if (sumSteps >= targetSteps) {
            console.log("Goal reached! Good job!");
            console.log(`${sumSteps - targetSteps} steps over the goal!`);
        } else {
            console.log(`${targetSteps - sumSteps} more steps to reach goal.`);
        }
    }

}


// walking([
//     "1000",
//     "1500",
//     "2000",
//     "6500",
// ]);

// walking([
//     "125",
//     "250",
//     "4000",
//     "30",
//     "2678",
//     "4682",  
// ]);

File: Python_Courses/Python_Basics/Preparation/05_While_Loop_Exercise/04_Walking.py
STEPS_GOAL = 10_000
GOING_HOME_COMMAND = "Going home"

total_steps = 0

while True:
    command = input()
    if command == GOING_HOME_COMMAND:
        steps_to_home = int(input())
        total_steps += steps_to_home
        break

    current_steps = int(command)
    total_steps += current_steps

    if total_steps >= STEPS_GOAL:
        break

is_goal_achieved = total_steps >= STEPS_GOAL
if is_goal_achieved:
    print("Goal reached! Good job!")
    print(f"{total_steps - STEPS_GOAL} steps over the goal!")
else:
    print(f"{STEPS_GOAL - total_steps} more steps to reach goal.")
